src/mesh/meshblock.cpp

```diff
@@ -171,20 +171,33 @@ void MeshBlock::Initialize(int igid, int ilid, LogicalLocation iloc,
   // removed, which can happen after dense-on-block for sparse
   // variables is in place and after we write "prolongate-in-one,"
   // this should be only for `Metadata::Independent`.
+
+  // TODO(LFR): vars_cc_ sets what variables are communicated across
+  // ranks during remeshing, so we want to be able to explicitly flag
+  // variables that need to be communicated using `Metadata::RemeshComm`.
+  // In the future, this needs to be cleaned up since `vars_cc_` is
+  // potentially used in the load balancing calculation, but not all
+  // variables that we may want to communicate are necessarily relevant
+  // to the cost per meshblock.
   const auto vars =
       real_container
-          ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost}, false)
+          ->GetVariablesByFlag(
+              {Metadata::Independent, Metadata::FillGhost, Metadata::RemeshComm}, false)
           .vars();
-  for (int n = 0; n < vars.size(); n++) {
+  for (int n = 0; n < vars.size(); ++n) {
     RegisterMeshBlockData(vars[n]);
   }
 
   if (pm->multilevel) {
+    const auto refine_vars =
+        real_container
+            ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost}, false)
+            .vars();
```

Comment:
Likely related to my question above: In which case you want to communicate a variable but not refine it?
I'm probably missing the big picture.

Reply:
We were running into issues in Riot where results differed when run on different numbers of ranks, which that seemed to be related to certain `dependent` variables not being initialized after blocks were moved between ranks. Presumably, by very carefully reconstituting the dependent variables, we could get bitwise agreement independent of number of ranks, but a simpler solution was to just communicate some dependent data. 

When refining, we still only want to refine `independent` variables.